kroma-validator/l2_output_submitter.go

```diff
@@ -421,9 +606,69 @@ func SubmitL2OutputTxData(abi *abi.ABI, output *eth.OutputResponse) ([]byte, err
 	)
 }
 
+// getNextValidatorAddress selects the appropriate contract and retrieves the next validator address.
+func (l *L2OutputSubmitter) getNextValidatorAddress(ctx context.Context) (common.Address, error) {
+	var contract interface {
+		NextValidator(opts *bind.CallOpts) (common.Address, error)
+	}
+
+	if l.isValManagerEnabled {
+		contract = l.valManagerContract
+	} else {
+		contract = l.valpoolContract
+	}
+
+	return contract.NextValidator(optsutils.NewSimpleCallOpts(ctx))
+}
+
+// startValidatorTx creates start validator tx candidate.
+// It sends the candidate to txCandidates channel to process validator's tx candidates in order.
+func (l *L2OutputSubmitter) startValidatorTx(data []byte) *txmgr.TxResponse {
+	gasTipCap, basefee, _, err := l.cfg.TxManager.SuggestGasPriceCaps(l.ctx)
+	if err != nil {
+		return &txmgr.TxResponse{
+			Receipt: nil,
+			Err:     fmt.Errorf("failed to get gas price info: %w", err),
+		}
+	}
+	gasFeeCap := txmgr.CalcGasFeeCap(basefee, gasTipCap)
+
+	to := &l.cfg.ValidatorManagerAddr
+	estimatedGas, err := l.cfg.L1Client.EstimateGas(l.ctx, ethereum.CallMsg{
+		From:      l.cfg.TxManager.From(),
+		To:        to,
+		GasFeeCap: gasFeeCap,
+		GasTipCap: gasTipCap,
+		Data:      data,
+	})
+	if err != nil {
+		return &txmgr.TxResponse{
+			Receipt: nil,
+			Err:     fmt.Errorf("failed to estimate gas: %w", err),
+		}
+	}
+
+	return l.cfg.TxManager.SendTxCandidate(l.ctx, &txmgr.TxCandidate{
+		TxData:   data,
+		To:       to,
+		GasLimit: estimatedGas * 3 / 2,
+	})
+
```

Comment:
We can just simplify it like below because gas estimation is done in `txManager` later. In `submitL2OutputTx`, we do gas estimation and multiply it with 1.5 to prevent tx failed during public round.
```suggestion
	return l.cfg.TxManager.SendTxCandidate(l.ctx, &txmgr.TxCandidate{
		TxData: data,
		To:     &l.cfg.ValidatorManagerAddr,
	})
```